Gordon360/Services/HousingService.cs

```diff
@@ -505,7 +506,18 @@ public ApartmentApplicationViewModel GetApartmentApplication(int applicationID,
         /// <returns>Array of ApartmentApplicationViewModels</returns>
         public ApartmentApplicationViewModel[] GetAllApartmentApplication()
         {
-            return _context.Housing_Applications.AsEnumerable().Select(a => GetApartmentApplication(a.HousingAppID, true)).ToArray();
+            var enumerable = _context.Housing_Applications.AsEnumerable();
+            var currentSession = Helpers.GetCurrentSession(_context);
+
+            // TO DO: Refactor Housing App so that the application itself is connected
+            //    to a session rather than the applicants!
+            enumerable = enumerable.Where(a => _context.Housing_Applicants.Where(
+                apl => apl.HousingAppID == a.HousingAppID
+            ).Any(
+                apl => apl.SESS_CDE == currentSession
+            ));
```

Comment:
Could make this cleaner by filtering on session code first:

```suggestion
            // TO DO: Refactor Housing App so that the application itself is connected
            //    to a session rather than the applicants!
            var currentApplicants = _context.Housing_Applicants.Where(
                a => a.SESS_CDE == currentSession
            );
            enumerable = enumerable.Where(a => currentApplicants.Any(
                apl => apl.HousingAppID == a.HousingAppID
            ));
```

Reply:
I do think the `Include` solution will be even cleaner. You'd have to call `.Include` on the `DBSet` before calling `AsEnumerable` though. Also, you shouldn't need to call `AsEnumerable` at all I don't think.